process/incubation.md

```diff
@@ -0,0 +1,138 @@
+
+#### Graduated Level
+
+To be accepted at Graduated level, a project must meet the Incubating level criteria plus:
+
+* Adopt all [SSWG best practices](#best-practices), as detailed below.
+* Have committers from at least two organizations.
+* Receive a supermajority vote from the SSWG to move to Graduation stage.
+
+### Ecosystem Index
+
+All projects and their respective levels will be listsed on the Swift Server Ecosystem index page. In cases where more than one project solves a particular problem (e.g., two similar database drivers), they will be ordered by popularity. The SSWG reserves the right to define a singular solution for critical building blocks, such as Logging or Metrics APIs, where consistency across the ecosystem is of a critical nature.
+
+The SSWG will meet every 6 month to review all projects, and it reserves the right to demote, archive, or remove projects that no longer fulfill minimal requirements. For example, a Graduated project that no longer receives regular updates or fails to address security concerns in timely fashion. Similarly, the SSWG reserves the right to remove or archive Pitches and Proposals that no longer receive updates.
+
+Changes to the Swift Server Ecosystem index page will be announced by the SSWG using the Swift Server forums.
+
+## Minimal Requirements
+
+* General
+  * Publicly accessible source managed by an SCM such as github.com or similar
+  * Adopt the [Swift Code of Conduct](https://swift.org/community/#code-of-conduct)
+  * [Developer Certificate of Origin](https://developercertificate.org) or a [Contributor License Agreement](https://en.wikipedia.org/wiki/Contributor_License_Agreement)
+* Ecosystem
+  * Uses SwiftPM
+  * Integrated with critical SSWG ecosystem building blocks, e.g., Logging and Metrics APIs, SwiftNIO for IO
+* Concurrency / IO
+  * Packages should be non-blocking (w/ async API) unless not possible (blocking C libs, etc)
+  * There should be as little (preferably no) wrapping of NIO as possible. Exposing NIO types directly will go a long way for making packages compatible.
+  * Blocking code should be wrapped in NonBlockingIOPool (like Vapor's SQLite package)
+* Testing, CI and Release
+  * Have unit tests for Linux
+  * CI setup, including testing PRs and the master branch
+  * Follow semantic versioning
+* Conventions and Style
+  * Error handling
+  * C wrapping vs native swift
+  * Code style guidelines
+  * Uses force unwraps and force tries only as preconditions, ie. conditions that the programmer regards as impossible or programmer error. All force tries/unwraps should come with a comment stating the reasons.
+* Longevity
+  * Must be a team of 2+ developers
+  * Must be from a team that has more than one public repository (or similar indication of experience)
+  * SSWG should have access / authorization to graduated repos in case of emergency
+* Licensing
+  * Apache 2
+
+## Best Practices
+
+* CI setup across all supported Swift and Linux versions
```

Comment:
We probably should clarify what "all supported Swift and Linux versions" mean, as these are things that change over time.

* When is a package expected to pick up the latest GM version of Swift?
* Are they expected to test against the latest development version of Swift?
* When is a Swift version no longer supported?  On Swift.org we keep links to all the GM releases, but we aren't recommending people use Swift 3 or even Swift 4 anymore, as there are newer releases
* When a new Linux distribution becomes supported, how quickly do they need to get that CI setup?  What if they *can't* setup CI for all the supported Linux distributions?

Reply:
BTW, these details are important for expectation setting not just for library owners but for the users of those libraries.  It instills confidence in their investments in these libraries to understand the level of support they have.

Reply:
@tkremenek wdyt about https://github.com/swift-server/sswg/pull/1/commits/cd18097ee16045ec2a98afe5755bb4bec49eba32